src/features/import-map.js

```diff
@@ -10,40 +10,68 @@
 import { baseUrl as pageBaseUrl, parseImportMap, resolveImportMap } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 
-var importMap, importMapPromise;
-if (typeof document !== 'undefined') {
-  const scripts = document.getElementsByTagName('script');
-  for (let i = 0; i < scripts.length; i++) {
-    const script = scripts[i];
-    if (script.type !== 'systemjs-importmap')
-      continue;
-
-    if (!script.src) {
-      importMap = parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl);
-    }
-    else {
-      importMapPromise = fetch(script.src)
-      .then(function (res) {
-        return res.json();
-      })
-      .then(function (json) {
-        importMap = parseImportMap(json, script.src);
-      });
+var acquiringImportMaps = true, importMapPromise;
+
+function retrieveImportMaps() {
```

Comment:
Instead of immediately processing the import maps when systemjs executes, we now wait until someone calls System.import for the first time.

An optimization would be to start fetching the import maps immediately, which is something I plan on implementing in this pull request.

Reply:
Yes, we should immediately start processing instead of deferring this.